.github/workflows/maven.yml

```diff
@@ -0,0 +1,39 @@
+name: Java CI with Maven
+
+on:
+  push:
+    branches: [ main ]
+  pull_request:
+    branches: [ main ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        java: [ 8, 11, 17 ]
+    steps:
+
+    - uses: actions/checkout@v2
+
+    - name: Set up JDK ${{ matrix.java }}
+      uses: actions/setup-java@v2
+      with:
+        java-version: ${{ matrix.java }}
+        distribution: temurin
+
+    - name: Cache local Maven repository
+      uses: actions/cache@v2
+      with:
+        path: ~/.m2/repository
+        key: ${{ runner.os }}-maven-${{ hashFiles('**/pom.xml') }}
+        restore-keys: |
+          ${{ runner.os }}-maven-
+
+    - name: Build with Maven
+      run: mvn -B package --file pom.xml
+
+    - uses: actions/upload-artifact@v2
```

Comment:
We don't need the Artifact upload. Instead, we could deploy the artifact to OSS Sonatype. It would also make sense to add guards to run the action only if it is this repository (`if: github.repository == 'mp911de/logstash-gelf'`) and run a verification only on forks.

See https://github.com/pgjdbc/r2dbc-postgresql/blob/main/.github/workflows/ci.yml for some ideas.

Reply:
Good idea! I have changed the PRt to upload to OSS Sonatype Snapshot if the according ENV variables are set.
For me this results in a HTTP 401 with my Sonatype credentials.
I think it's better to check against the env vars than to hard-code the repo. What do you think?